plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/ui/wizard/MessageStoreCreationWizard.java

```diff
@@ -365,6 +379,56 @@ private String getTemplateContent(){
 					}
 				}
 			}
+		} else if (messageStoreModel.getMessageStoreType() == MessageStoreType.RESEQUENCE) {
+			className = RESEQUENCE_MS_FQN;
+			if (StringUtils.isNotBlank(messageStoreModel.getResequenceDatabaseTable())) {
+				parameters.put(STORE_RESEQUENCE_TABLE, messageStoreModel.getResequenceDatabaseTable());
+			}
+			if (StringUtils.isNotBlank(messageStoreModel.getResequenceEnableProducerGuaranteedDelivery())) {
+				parameters.put(STORE_RESEQUENCE_PRODUCER_GUARANTEED_DELIVERY_ENABLE, messageStoreModel.getResequenceEnableProducerGuaranteedDelivery());
+			}
+			if (StringUtils.isNotBlank(messageStoreModel.getResequenceFailoverMessageStore())) {
+				parameters.put(STORE_RESEQUENCE_FAILOVER_MESSAGE_STORE_NAME, messageStoreModel.getResequenceFailoverMessageStore());
+			}
+			if (StringUtils.isNotBlank(messageStoreModel.getResequencePollingCount())
+					&& StringUtils.isNumeric(messageStoreModel.getResequencePollingCount())) {
+				parameters.put(STORE_RESEQUENCE_POLLING_COUNT, messageStoreModel.getResequencePollingCount());
+			}
+			if (StringUtils.isNotBlank(messageStoreModel.getResequenceXpath())) {
+				SynapseXPath xpath;
+				try {
+					xpath = new SynapseXPath(messageStoreModel.getResequenceXpath());
+					Map<String,String> maps = messageStoreModel.getResequenceXpathNamespaces(); //Add namespaces
+					for(String key:maps.keySet()) {
+						xpath.addNamespace(key,maps.get(key));
+					}
+					parameters.put(STORE_RESEQUENCE_XPATH, xpath);
+				} catch (JaxenException e) {
+					log.error("Cannot convert xpath", e);
+				}
+			}
+			// Switch between connection pool and datasource
+			String resequenceConnectionInformation = messageStoreModel.getResequenceConnectionInformation();
+			if (StringUtils.isNotBlank(resequenceConnectionInformation)) {
+				if (JDBCConnectionInformationType.POOL.toString().equals(resequenceConnectionInformation)) {
+					if (StringUtils.isNotBlank(messageStoreModel.getResequenceDriver())) {
+						parameters.put(STORE_RESEQUENCE_DRIVER, messageStoreModel.getResequenceDriver());
+					}
+					if (StringUtils.isNotBlank(messageStoreModel.getResequenceURL())) {
+						parameters.put(STORE_RESEQUENCE_CONNECTION_URL, messageStoreModel.getResequenceURL());
+					}
+					if (StringUtils.isNotBlank(messageStoreModel.getResequenceUser())) {
+						parameters.put(STORE_RESEQUENCE_USERNAME, messageStoreModel.getResequenceUser());
+					}
+					if (StringUtils.isNotBlank(messageStoreModel.getResequencePassword())) {
+						parameters.put(STORE_RESEQUENCE_PASSWORD, messageStoreModel.getResequencePassword());
+					}
+				} else if (JDBCConnectionInformationType.CARBON_DATASOURCE.toString().equals(resequenceConnectionInformation)) {
+					if (StringUtils.isNotBlank(messageStoreModel.getResequenceDatasourceName())) {
```

Comment:
we can merge these two else if and if blocks to one

Reply:
Fixed with 7c27579